configuration/NI.Analysis.targets

```diff
@@ -0,0 +1,107 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <!-- TODO: This location of 'content' needs to get updated once we build our package if we decide to put the files elsewhere, outside of a 'content' directory -->
+    <CodeAnalysisRuleSetDirectory>$(PkgNI_DotNet_Analyzers)\content</CodeAnalysisRuleSetDirectory>
+    <NuGetPackagesDirectory>$(NuGetPackageRoot)</NuGetPackagesDirectory>
+  </PropertyGroup>
+
+	<PropertyGroup>
+		<!-- this would more correctly be: assemblyname contains .Test. We have an issue with test utilities projects, and test.testutilities projects. -->
+		<IsNITestProject Condition="'$(IsNITestProject)' != 'True'">$(MSBuildProjectName.StartsWith("Tests."))</IsNITestProject>
+		<IsNITestProject Condition="'$(IsNITestProject)' != 'True'">$(MSBuildProjectName.EndsWith(".Tests"))</IsNITestProject>
+		<IsNITestProject Condition="'$(IsNITestProject)' != 'True'">$(MSBuildProjectName.StartsWith("!Tests."))</IsNITestProject>
+		<IsNITestUtilitiesProject Condition="'$(IsNITestUtilitiesProject)' != 'True'">$(MSBuildProjectName.StartsWith("TestUtilities."))</IsNITestUtilitiesProject>
+		<IsNITestUtilitiesProject Condition="'$(IsNITestUtilitiesProject)' != 'True'">$(MSBuildProjectName.StartsWith("!TestUtilities."))</IsNITestUtilitiesProject>
+		<!-- The following is needed for misnamed TestUtilities that contain xaml because the generated project for the second compile appends random characters to the end of the project name -->
+		<IsNITestUtilitiesProject Condition="'$(IsNITestUtilitiesProject)' != 'True' and '$(IsNITestProject)' != 'True'">$(MSBuildProjectName.Contains(".TestUtilities"))</IsNITestUtilitiesProject>
+		<ProjectNameContainsTests>$(MSBuildProjectName.Contains("Tests"))</ProjectNameContainsTests>
+
+		<CodeAnalysisRuleSet>$(CodeAnalysisRuleSetDirectory)\NI.ruleset</CodeAnalysisRuleSet>
+		<CodeAnalysisRuleSet Condition="'$(IsNITestProjectCondition)' == 'True'">$(CodeAnalysisRuleSetDirectory)\NI.Tests.ruleset</CodeAnalysisRuleSet>
+		<CodeAnalysisRuleSet Condition="'$(IsNITestUtilitiesProjectCondition)' == 'True'">$(CodeAnalysisRuleSetDirectory)\NI.TestUtilities.ruleset</CodeAnalysisRuleSet>
+	</PropertyGroup>
+
+  <PropertyGroup>
+    <!-- Enable Roslyn analyzers if they haven't been explicitly disabled, we're using Roslyn for CA/SA, the code needs to be validated -->
+    <EnableRoslynAnalyzers>True</EnableRoslynAnalyzers>
+    <IncludeStyleAnalyzers>True</IncludeStyleAnalyzers>
+    <IncludeCodeAnalyzers>True</IncludeCodeAnalyzers>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <NI1006_BannedMethods>$(SolutionDir)\BannedMethods.xml</NI1006_BannedMethods>
```

Comment:
I removed Literal Exemptions from this file since it won't be used anymore. I changed BannedMethods to be included from $SolutionDir instead of anything relative to this package.